pages/ActivityStreams/Overview.md

```diff
@@ -33,22 +33,347 @@ These deviations from the standard will be thoroughly documented here.
 ## Minimum Subset
 
 At minimum, all clients implementing the DSNP must support presentation for the following message types from the Activity Streams vocabulary: `"Link"`, `"Note"` and `"Person"`.
-It is also strongly encouraged to support `"Audio"`, `"Image"`, `"Mention"`, `"Profile"`, `"Tombstone"` and `"Video"` where ever possible.
+It is also strongly encouraged to support `"Audio"`, `"Image"`, `"Profile"` and `"Video"` where ever possible.
 Clients may ignore messages that they do not support.
 
+### Required Activity Types
+
+#### Link
+
+Link type objects must be supported to represent linked content posted by users.
+Link objects must have a `"type"` field with a value of `"Link"` and a `"href"` field with a valid URL string value.
+Clients must at least support valid URLs with a `"https://"` protocol, but clients may choose to ignore any other protocols as they see appropriate for the security of users.
+Clients may also choose to ignore link objects with IP address hostnames or invalid domain names.
+
+Link objects may also include a `"mediaType"` field with a standard [MIME type](https://www.iana.org/assignments/media-types/media-types.xhtml) string value representing the type of content at the linked URI.
+
+For example, the following would be a valid link object:
+
+```json
+{
+  "@context": "https://www.w3.org/ns/activitystreams",
+  "type": "Link",
+  "href": "https://dsnp.org"
+}
+```
+
+And implementers may choose to ignore the following link objects:
+
+```json
+{
+  "@context": "https://www.w3.org/ns/activitystreams",
+  "type": "Link",
+  "href": "ipfs://bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfyavhwq/wiki/Vincent_van_Gogh.html"
+}
+```
+
+```json
+{
+  "@context": "https://www.w3.org/ns/activitystreams",
+  "type": "Link",
+  "href": "http://192.168.1.1/router-setup"
+}
+```
+
+#### Note
+
+Note type objects must be supported to represent text content posted by users.
+Note objects must have a `"type"` field with a value of `"Note"` and a `"content"` field with a string value.
+Clients should support display of content with unicode text as possible in their given browser or operating system.
+Clients may also support display of [Markdown](https://daringfireball.net/projects/markdown/) or [BBCode](https://www.bbcode.org) as appropriate.
+If a note contains Markdown, BBCode or some other formatting, it is recommended that an `"mediaType"` field be added to the object with a [MIME type](https://www.iana.org/assignments/media-types/media-types.xhtml) string value representing the type of encoding present, i.e. `"text/markdown"`.
```

Comment:
I think my reasoning for this being a recommendation instead of a requirement is because it's not really enforceable as a requirement. There's no way I could distinguish markdown text in a note's content that doesn't have a `mediaType` field from plain text that doesn't need a `mediaType` field, so I wouldn't be able to mark those messages as invalid and ignore them.

I suppose we could say it's a requirement here, but there's no validation function we could write to back it up.

Reply:
On second thought, I've updated this to be a requirement.

Reply:
I know I am late to this, but could we make mediaType a "may", but say if not present, it must be interpreted as "text/plain" (or some other default type)? That gives a well defined path for handling things if mediaType is absent, vs the message just being invalid.